native/src/main/java/io/ballerina/stdlib/etl/nativeimpl/EtlSecurity.java

```diff
@@ -0,0 +1,143 @@
+        }
+        return encryptedDataset;
+    }
+
+    public static Object decryptData(BArray dataset, BArray fieldNames, BArray key, BTypedesc returnType) {
+        for (int i = 0; i < fieldNames.size(); i++) {
+            if (!isFieldExist(dataset, fieldNames.getBString(i))) {
+                return ErrorUtils.createFieldNotFoundError(fieldNames.getBString(i));
+            }
+        }
+        BArray decryptedDataset = initializeBArray(returnType);
+        for (int i = 0; i < dataset.size(); i++) {
+            if (TypeUtils.getType(dataset.get(i)).getTag() != TypeTags.RECORD_TYPE_TAG) {
+                continue;
+            }
+            BMap<BString, Object> data = (BMap<BString, Object>) dataset.get(i);
+            BMap<BString, Object> decryptedData = initializeBMap(returnType);
+            BString[] keys = data.getKeys();
+            for (BString keyField : keys) {
+                if (contains(fieldNames, keyField)) {
+                    Object decryptedBytes = FromBase64
+                            .fromBase64(StringUtils.fromString(data.get(keyField).toString()));
+                    if (TypeUtils.getType(decryptedBytes).getTag() != TypeTags.ARRAY_TAG) {
+                        ErrorUtils.createDecryptionError();
+                    }
+                    Object decryptedValue = Decrypt.decryptAesEcb((BArray) decryptedBytes, key,
+                            "PKCS5");
+                    if (TypeUtils.getType(decryptedValue).getTag() != TypeTags.ARRAY_TAG) {
+                        ErrorUtils.createDecryptionError();
+                    }
+                    decryptedData.put(keyField, StringUtils
+                            .fromString(new String((((BArray) decryptedValue).getBytes()), StandardCharsets.UTF_8)));
+                } else {
+                    decryptedData.put(keyField, data.get(keyField));
+                }
+            }
+            decryptedDataset.append(decryptedData);
+        }
+        return decryptedDataset;
+    }
+
+    public static Object maskSensitiveData(Environment env, BArray dataset, BString maskCharacter,
+            BTypedesc returnType) {
+        BArray mergedResult = initializeBArray(returnType);
+        for (int i = 0; i < dataset.size(); i += 200) {
+            int end = Math.min(i + 200, dataset.size());
+            BArray chunk = dataset.slice(i, end);
+            Object[] args = new Object[] { chunk, maskCharacter };
+            Object clientResponse = env.getRuntime().callFunction(env.getCurrentModule(), MASK_SENSITIVE_DATA, null,
+                    args);
+            Object chunkResult = processResponseToBArray(clientResponse, returnType);
+            if (TypeUtils.getType(chunkResult).getTag() != TypeTags.ARRAY_TAG) {
+                return chunkResult;
+            }
+            for (int j = 0; j < ((BArray) chunkResult).size(); j++) {
+                mergedResult.append(((BArray) chunkResult).get(j));
+            }
+            ;
```

Comment:
Check the formatting and why there's a semi-colon here?

```suggestion
            ;
```